File: Robot-Return-to-Origin/RRtO.rs
pub fn judge_circle(moves: String) -> bool {
    let mut h = 0;
    let mut c = 0;
    for cc in moves.chars() {
        match cc {
            'U' => {
                h += 1;
            }
            'D' => {
                h -= 1;
            }
            'L' => {
                c -= 1;
            }
            'R' => c += 1,
            _ => unreachable!(),
        }
    }
    h == 0 && c == 0
}

fn main() {
    assert!(judge_circle("UD".to_string()));
    assert!(!judge_circle("LL".to_string()));
}

File: Robot-Return-to-Origin/RRtO.ml
let judge_circle moves =
  let h = ref 0 and c = ref 0 in
  moves |> String.to_seq |> List.of_seq
  |> List.iter (fun cc ->
         match cc with
         | 'U' -> h := !h + 1
         | 'D' -> h := !h - 1
         | 'L' -> c := !c - 1
         | 'R' -> c := !c + 1
         | _ -> ());
  !h == 0 && !c == 0

let main =
  assert (judge_circle "UD");
  assert (not (judge_circle "LL"))
